src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/CustomValidationAttribute.cs

```diff
@@ -184,12 +184,7 @@ protected override ValidationResult IsValid(object value, ValidationContext vali
             }
             catch (TargetInvocationException tie)
             {
-                if (tie.InnerException != null)
-                {
-                    throw tie.InnerException;
-                }
-
-                throw;
+                throw tie.InnerException;
```

Comment:
Do you know for sure that InnerException will always be non-null in all circumstances?